wrapper.go

```diff
@@ -0,0 +1,125 @@
+package wflambda
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/rcrowley/go-metrics"
+	"reflect"
+	"time"
+)
+
+type lambdaHandlerFunction func(context.Context, json.RawMessage) (interface{}, error)
+
+var (
+	server                  string
+	authToken               string
+	reportStandardMetrics   bool
+	csCounter               metrics.Counter
+	csEventCounter          metrics.Counter
+	invocationsCounter      metrics.Counter
+	invocationEventCounter  metrics.Counter
+	errCounter              metrics.Counter
+	errEventCounter         metrics.Counter
+	dGauge                  metrics.Gauge
+	coldStart               = true
+)
+
+ // Returns the Wavefront Lambda wrapper. The wrapper collects aws lambda's
+ //   standard metrics and reports it directly to the specified wavefront url. It
+ //   requires the following Environmental variables to be set:
+ //   1.WAVEFRONT_URL : https://<INSTANCE>.wavefront.com
+ //   2.WAVEFRONT_API_TOKEN : Wavefront API token with Direct Data Ingestion permission
+ //   3.IS_REPORT_STANDARD_METRICS : Set to False to not report standard lambda
+ //                                 metrics directly to wavefront.
+
+func Wrapper(lambdaHandler interface{}) interface{} {
+  // Validate wrapper environment variables.
+  reportStandardMetrics = getAndValidateLambdaEnvironment()
+	// Check if lambdaHandler is a valid handler.
+  lambdaHandlerTakesContext, err := validateLambdaHandler(lambdaHandler)
+  if err != nil {
+		return lambdaErrorHandler(err)
+	}
+  handlerType := reflect.TypeOf(lambdaHandler)
+  handlerValue := reflect.ValueOf(lambdaHandler)
+
+	//Returns a wavefrontLambda wrapper function with standard lambda metrics.
+	return func(ctx context.Context, payload json.RawMessage) (response interface{}, lambdaHandlerError error) {
```

Comment:
It'd be cleaner to move this logic into a named function outside of the Wrapper func and then just return that func.
Also remove the named return parameters `response` and `lambdaHandlerError.  That is only useful if you have a return statement without arguments, which you don't seem to have: https://tour.golang.org/basics/7

Example:
```
func foo(ctx context.Context, payload json.RawMessage) (interface{}, error) {
 // do something
}

func Wrapper(...) {
    return foo
}
```